src/components/list/list.js

```diff
@@ -205,8 +205,9 @@ function mdListItemDirective($mdAria, $mdConstant, $timeout) {
 
         if (!hasClick && !proxies.length) {
           firstChild.addEventListener('keypress', function(e) {
-            if (e.target.nodeName != 'INPUT') {
-              if (e.keyCode == $mdConstant.KEY_CODE.SPACE) {
+            if (e.target.nodeName != 'INPUT' && e.target.nodeName != 'TEXTAREA') {
+              var keyCode = e.which || e.keyCode;
```

Comment:
For compatibility with Firefox, which returns 0 for `e.keyCode` here. This fallback is more than once in the material source. See my comments in the corresponding issue (#2668).